backend/src/controllers/Product/index.js

```diff
@@ -0,0 +1,120 @@
+import { productRepository } from '#services';
+import { requireRole } from '#middleware';
+
+// paginated
+// not to be confused with catalogue!
+export async function list(req, res, next) {
+    try {
+        await requireRole(
```

Comment:
trying to make this a bit more common to reduce code-dup'

should be helpful for everyone else too